UnchainedLauncherCore/src/API/A2S.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net;
+using System.Net.Sockets;
+using System.Buffers.Binary;
+
+namespace UnchainedLauncher.Core.API
+{
+    public record A2S_INFO (
+        byte ProtocolVersion, 
+        String Name,
+        String Map,
+        String Folder,
+        String Game, 
+        UInt16 GameID, 
+        byte Players,
+        byte MaxPlayers,
+        byte Bots, 
+        ServerType ServerType, 
+        Environment Environment, 
+        bool IsPublic, 
+        bool Vac
+        );
+
+    public enum ServerType : byte
+    {
+        DEDICATED = (byte)'D',
+        NONDEDICATED = (byte)'L',
+        PROXY = (byte)'P'
+    }
```

Comment:
TIL that the char type in C# is 16 bits (to support UTF-16).

Reply:
To clarify what I mean, I was confused by the `byte` cast, but it makes sense in the light of chars being 16 bit